File: 1759-count-number-of-homogenous-substrings/1759-count-number-of-homogenous-substrings.cpp
class Solution {
public:
    int countHomogenous(string s) {
        // Using sliding window
        const int mod = 1e9+7;
        int i = 0,j = 0;
        int n = s.size();
        int ans = 0;
        //sliding window
        while(j<n)
        {
            
            while(s[i] != s[j])
            {
                i++;
            }
            
            if(s[i] == s[j])
            {
                ans =  (ans + (j-i+1))%mod;
            }
            
            j++;
        }
        
        return ans;
    }
};

File: 1759-count-number-of-homogenous-substrings/1759-count-number-of-homogenous-substrings.java
class Solution {
    public int countHomogenous(String s) {
        
        int mod = 1000000007;
        
        int len = 0;
        int res = 0;
        for(int i=0; i< s.length(); i++){
            if(i>0 && s.charAt(i) == s.charAt(i-1)){
                len +=1;
                
            }else{
                len = 1;
            }
            
            res = (res + len) % mod;
            
        }
        
        return res;
    }
}